new-lamassu-admin/src/pazuz/apollo/Provider.js

```diff
@@ -0,0 +1,57 @@
+import { ApolloProvider } from '@apollo/react-hooks'
+import { InMemoryCache } from 'apollo-cache-inmemory'
+import { ApolloClient } from 'apollo-client'
+import { ApolloLink } from 'apollo-link'
+import { onError } from 'apollo-link-error'
+import { HttpLink } from 'apollo-link-http'
+import React from 'react'
+import { useHistory, useLocation } from 'react-router-dom'
+
+const URI =
+  process.env.NODE_ENV === 'development' ? 'https://localhost:8070' : ''
+
+const getClient = (history, location) =>
+  new ApolloClient({
+    link: ApolloLink.from([
+      onError(({ graphQLErrors, networkError }) => {
+        if (graphQLErrors)
+          graphQLErrors.forEach(({ message, locations, path, extensions }) => {
+            if (extensions?.code === 'UNAUTHENTICATED') {
+              if (location.pathname !== '/404') history.push('/404')
```

Comment:
can't we have this inside a single if?

Reply:
Wouldn't that lead to repeated code? As there's code left to execute inside the first `if` statement.

Reply:
you're right, missed it!